src/NextGen/dataRepository.ts

```diff
@@ -4,6 +4,7 @@ import { BaseMPMessage } from './core/baseMPMessage';
 export class MParticleDataRepository {
     public internalState: BaseMPMessage[] = [];
 
+    // QUESTION: Shouldn't this be of type
     public insertCommerceEvent(data: BaseMPMessage): void {
```

Comment:
Yeah I'm not sure I understand the code here. Shouldn't this be a general insert for all kinds of events? or should there be one for non-commerce events as well?

Reply:
I agree, but in the android code, it's [`insertCommerceDTO`](https://github.com/mParticle/mparticle-android-sdk/pull/367/files#diff-b478a78e10bff58c9c8aeb10ee428976f0c886ab1db3e7b6c15b110e8e243400R14), so we may need to get @markvdouw to clarify intent here.

Reply:
It was just an example. If the strategy is incharge of mapping, we can save a generic one.